dex-it/src/test/scala/com/wavesplatform/it/sync/MatcherRestartTestSuite.scala

```diff
@@ -1,33 +1,61 @@
 package com.wavesplatform.it.sync
 
+import com.wavesplatform.dex.api.http.entities.HttpOrderStatus
 import com.wavesplatform.dex.api.http.entities.HttpOrderStatus.Status
 import com.wavesplatform.dex.domain.order.{Order, OrderType}
 import com.wavesplatform.it.MatcherSuiteBase
+import org.scalacheck.Gen
 
 class MatcherRestartTestSuite extends MatcherSuiteBase {
 
   override protected def beforeAll(): Unit = {
     super.beforeAll()
     broadcastAndAwait(IssueEthTx)
+    broadcastAndAwait(IssueWctTx)
   }
 
   "check order execution" - {
     "make order and after matcher's restart try to cancel it" in {
       // Alice places sell order
       val aliceOrder = mkOrder(alice, ethWavesPair, OrderType.SELL, 500, 2.waves * Order.PriceConstant)
+      val bobsOrderToCancel = mkBobOrder()
+      val bobsOrderToFulfil = mkBobOrder()
+      val bobsOrders = (1 to 10).map(_ => mkBobOrder())
+
+      markup("place Alice order")
       placeAndAwaitAtDex(aliceOrder)
 
+      markup("place Bob's order and cancel it")
+      placeAndAwaitAtDex(bobsOrderToCancel)
+      dex1.api.cancelOrderById(bobsOrderToCancel)
+      dex1.api.waitForOrderStatus(bobsOrderToCancel, Status.Cancelled)
+
+      markup("place Bob's order and fulfil it")
+      placeAndAwaitAtDex(bobsOrderToFulfil)
+      val aliceSecondOrder = mkOrder(alice, wctWavesPair, OrderType.BUY, bobsOrderToFulfil.amount, 2.waves * Order.PriceConstant)
+      placeAndAwaitAtDex(aliceSecondOrder, HttpOrderStatus.Status.Filled)
+      eventually {
+        dex1.api.waitForOrderStatus(bobsOrderToFulfil, Status.Filled)
+      }
+
+      markup("place other Bob's orders")
+      bobsOrders.foreach(v => placeAndAwaitAtDex(v))
+
       // Check that order is correct
+      markup("check Alice order")
```

Comment:
Let's use `print` from https://github.com/wavesplatform/matcher/blob/master/dex-it-common/src/main/scala/com/wavesplatform/dex/it/test/InformativeTestStart.scala#L29 instead. It writes a log to all files.